test/behaviour/typeql/TypeQLSteps.java

```diff
@@ -159,6 +169,31 @@ public void typeql_match(String typeQLQueryStatements) {
         }
     }
 
+    private void testProcedurePermutations(TypeQLMatch typeQLQuery) {
+        Disjunction disjunction = Disjunction.create(typeQLQuery.conjunction().normalise());
+        tx().logic().typeInference().applyCombination(disjunction);
+        Set<Identifier.Variable.Retrievable> filter = iterate(typeQLQuery.modifiers().filter())
+                .map(unboundVar -> Identifier.Variable.of(unboundVar.reference().asReferable()))
+                .filter(Identifier::isRetrievable).map(Identifier.Variable::asRetrievable)
+                .toSet();
+        for (Conjunction conjunction : disjunction.conjunctions()) {
+            GraphTraversal.Thing traversal = conjunction.traversal(filter);
+            // TODO we expect there to be be only 1 structure in the future
+            List<Structure> structures = traversal.structure().asGraphs();
+            for (Structure structure : structures) {
+                if (structure.vertices().size() == 1)  continue;
+                List<GraphProcedure> procedurePermutations = ProcedurePermutations.generate(structure);
+                Set<VertexMap> answers = procedurePermutations.get(0).iterator(tx().concepts().graph(),
+                        traversal.parameters(), traversal.filter()).toSet();
+                for (int i = 1; i < procedurePermutations.size(); i++) {
+                    Set<VertexMap> permutationAnswers = procedurePermutations.get(i).iterator(tx().concepts().graph(),
+                            traversal.parameters(), filter).toSet();
+                    assertEquals(answers, permutationAnswers);
+                }
```

Comment:
because right now the `Structure` can split into disconnected sub-graphs, we just handle them separately and check all permutations' answer sets are equal to the first. This will all be unified into a single loop once the `Structures` do not partition themselves into multiple graphs.